### ISSUE TITLE
when i check or uncheck a supplier the earlier datas will remove from input

### ISSUE THREAD

**Issue by Flo340** [PrestaShop/PrestaShop#28425]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

I have a problem in the product BO page, when I check or uncheck a supplier the previous data will be removed from the entry.
**_The bug is present only if the product has variations_**

Ref supplier Ok 
![image](https://user-images.githubusercontent.com/104994978/166940391-e0a4d3a7-ec3c-412d-b035-efcc7fcc3612.png)

i uncheck a just a 1 supplier and all references disappear from all suppliers even suppliers still active

![image](https://user-images.githubusercontent.com/104994978/166940474-3b9d1349-2fc1-4432-a4a3-e70ffd454f68.png)

### Expected behavior

Normally only the data of the deactivated provider must be emptied, not the others

### Steps to reproduce

1 Go to a product where there are variations
2 Add more than 2 suppliers
3 Add the references for each supplier and each combination
4 Deactivate a provider

### PrestaShop version(s) where the bug happened

1.7.8

### PHP version(s) where the bug happened

7.3

**Comment by HanaRebaiQA** [comment]
Hello @Flo340 

Thanks for your report.

Your issue will be fixed in this [PR](https://github.com/PrestaShop/PrestaShop/pull/27353) 
So, i'll close this issue.

Do not hesitate to subscribe on this PR to be notified when it's fixed.

Thank you!

**Comment by Flo340** [comment]
> Hello @Flo340
> 
> Thanks for your report.
> 
> Your issue will be fixed in this [PR](https://github.com/PrestaShop/PrestaShop/pull/27353) So, i'll close this issue.
> 
> Do not hesitate to subscribe on this PR to be notified when it's fixed.
> 
> Thank you!

Hello
You have to change the 97 files, right?
https://github.com/PrestaShop/PrestaShop/pull/27353

**Comment by HanaRebaiQA** [comment]
@Flo340 I suppose if you can wait when the PR will be merged.

**Comment by Flo340** [comment]
> @Flo340 I suppose if you can wait when the PR will be merged.

We made the changes but under PS 1.7.8.5 it does not work the codes are different
are you sure that the bug is solved for 1.7.8.5 with this patch ? 
Thank you

**Comment by tom-combet** [comment]
While waiting for the product page v2, we can fix the bug by changing this line:

https://github.com/PrestaShop/PrestaShop/blob/dc8dc1faef7185ad30fc8dcb0653524d9bfb7c82/classes/ProductSupplier.php#L105

to:

```php
if ($item->id_product_attribute > 0 && $item->id_supplier == $this->id_supplier) {
```
